File: C sharp/StringCalculator/StringCalculatorTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace BowlingGameTest.StringCalculator
{
    [TestClass]
    public class StringCalculatorTest
    {
        private StringCalculator sc;

        [TestInitialize]
        public void setUp()
        {
            sc = new StringCalculator();
        }
        [TestMethod]
        public void Test_Add_Empty()
        {     
            Assert.AreEqual(0,sc.Add(""));
        }
        [TestMethod]
        public void Test_Add_One_Number() 
        {
            int n=20;
            Assert.AreEqual(n,sc.Add(n+""));
        }
        [TestMethod]
        public void Test_Add_Two_Number() 
        {
            Assert.AreEqual(6, sc.Add("1\n2,3"));
        }
        [TestMethod]
        public void Test_Add_Other() 
        {
            Assert.AreEqual(3, sc.Add("//;\n1;2"));
        }
        [TestMethod]
        public void Test_Negative_Number() 
        {
            //Assert.AreNotEqual(-1, sc.Add("-1"));
            //sc.Add("-1");
        }
        [TestMethod]
        public void Test_Greater_1000_Number() 
        {
            Assert.AreEqual(0, sc.Add("\\;1001,0\n2000"));
        }
    }
}

File: Java/Kata/src/StringCalculatorTest.java
import static org.junit.Assert.*;
import junit.framework.TestCase;

import org.junit.Test;


public class StringCalculatorTest extends TestCase{
	
	private StringCalculator sc;
	
	protected void setUp()
	{
		sc = new StringCalculator();
	}
	
	@Test
	public void test_add_Empty() {
		assertEquals(0,sc.add(""));
	}
	
	@Test
	public void test_add_One_Number()
	{
		assertEquals(1,sc.add("1"));
	}
	@Test
	public void test_add_Two_Numbers()
	{
		assertEquals(4,sc.add("1\n1,2"));
	}
	@Test
	public void test_add_ChangeDelimiter()
	{
		assertEquals(3,sc.add("//;\n1;2"));
	}
}
